caluma/caluma_workflow/migrations/0029_task_continue_async.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 3.2.12 on 2022-03-07 16:40
+
+from django.db import migrations, models
+
+
+def set_default_continue_async(apps, schema_editor):
+    apps.get_model("caluma_workflow", "Task").objects.filter(
+        is_multiple_instance=True
+    ).update(continue_async=False)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("caluma_workflow", "0028_redoable_field"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="historicaltask",
+            name="continue_async",
+            field=models.BooleanField(
+                blank=True,
+                default=None,
+                help_text="Whether to continue the flow if the multiple instance work item has ready siblings",
+                null=True,
+            ),
+        ),
+        migrations.AddField(
+            model_name="task",
+            name="continue_async",
+            field=models.BooleanField(
+                blank=True,
+                default=None,
+                help_text="Whether to continue the flow if the multiple instance work item has ready siblings",
+                null=True,
+            ),
+        ),
+        migrations.RunPython(set_default_continue_async, migrations.RunPython.noop),
```

Comment:
This should probably have a follow-up step where the `null=True` is removed in favour of `default=False`, no? Otherwise, we wouldn't have the need to set it explicitly to `False` here anyway.

If I remember correctly, this might require an additional migration file to avoid overzealous optimisation by Django.

Reply:
No, `null=True` is intended since a default to `False` would be missleading for tasks that are not multiple instance. To ensure the default behaviour it's explicitly set to `False` for multiple instance tasks.

Reply:
Not sure I'm buying it.. the behaviour would be exactly the same (The `not task.continue_async` check), it's just not having an effect in non-multiple-instance tasks.

I'm also not sure it would really be misleading. An explanatory comment would probably be a good idea in either case though

Reply:
Sure, the behaviour there would be the same. However I think that having a value set to `False` when it is not relevant for that object is less clean than setting it to `None`. I don't really care enough about this to discuss this any further so I'll change it.